File: comic/src/app/model/chapter.ts

export class Chapter{
  constructor(private _id: number,
              private _name: string,
              private _comic_id: number,
              private _content: string,
              private _date_created: string) {
  }


  get id(): number {
    return this._id;
  }

  set id(value: number) {
    this._id = value;
  }

  get name(): string {
    return this._name;
  }

  set name(value: string) {
    this._name = value;
  }

  get comic_id(): number {
    return this._comic_id;
  }

  set comic_id(value: number) {
    this._comic_id = value;
  }

  get content(): string {
    return this._content;
  }

  set content(value: string) {
    this._content = value;
  }

  get date_created(): string {
    return this._date_created;
  }

  set date_created(value: string) {
    this._date_created = value;
  }
}

File: DemoComicWebApp/src/main/java/com/example/webcomic/project/demo/model/Chapter.java
package com.example.webcomic.project.demo.model;

import java.util.Calendar;
import java.util.List;

import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.OneToMany;

import org.springframework.beans.factory.annotation.Autowired;

import com.example.webcomic.project.demo.repository.ComicRepository;
@Entity
public class Chapter {
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private long id;
	 @ManyToOne(fetch = FetchType.LAZY, optional = false)
	    @JoinColumn(name = "comic_id", nullable = false)
	private Comic comic;
	private String name;
	private Calendar date_created;
	private String content;
	@OneToMany(mappedBy = "chapter", fetch = FetchType.LAZY,
            cascade = CascadeType.ALL)
	private List<Comment> listComment;
	
	public Chapter() {
		super();
	}



	public Chapter(long id, Comic comic, String name, Calendar date_created, String content,
			List<Comment> listComment) {
		super();
		this.id = id;
		this.comic = comic;
		this.name = name;
		this.date_created = date_created;
		this.content = content;
		this.listComment = listComment;
	}

	

	public Chapter(long id, String name, Calendar date_created, String content, List<Comment> listComment) {
		super();
		this.id = id;
		this.name = name;
		this.date_created = date_created;
		this.content = content;
		this.listComment = listComment;
	}



	public long getId() {
		return id;
	}

	public void setId(long id) {
		this.id = id;
	}

//	public Comic getComic() {
//		return comic;
//	}
//
	public void setComic(Comic comic) {
		this.comic = comic;
	}
//
	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public Calendar getDate_created() {
		return date_created;
	}

	public void setDate_created(Calendar date_created) {
		this.date_created = date_created;
	}

	public String getContent() {
		return content;
	}

	public void setContent(String content) {
		this.content = content;
	}

	public List<Comment> getListComment() {
		return listComment;
	}

	public void setListComment(List<Comment> listComment) {
		this.listComment = listComment;
	}
	
	
}
